src/store/src/Bridge/SurrealDB/Store.php

```diff
@@ -0,0 +1,182 @@
+        private readonly string $vectorFieldName = '_vectors',
+        private readonly string $strategy = 'cosine',
+        private readonly int $embeddingsDimension = 1536,
+    ) {
+    }
+
+    public function add(VectorDocument ...$documents): void
+    {
+        $authenticationToken = $this->authenticate([]);
+
+        foreach ($documents as $document) {
+            $this->request('POST', \sprintf('key/%s', $this->table), $this->convertToIndexableArray($document), [
+                'Surreal-NS' => $this->namespace,
+                'Surreal-DB' => $this->database,
+                'Authorization' => \sprintf('Bearer %s', $authenticationToken),
+            ]);
+        }
+    }
+
+    public function query(Vector $vector, array $options = [], ?float $minScore = null): array
+    {
+        $authenticationToken = $this->authenticate($options);
+
+        $vectors = json_encode($vector->getData());
+
+        $results = $this->request('POST', 'sql', \sprintf(
+            'SELECT id, %s, _metadata, vector::similarity::%s(%s, %s) AS distance FROM %s WHERE %s <|2|> %s;',
+            $this->vectorFieldName, $this->strategy, $this->vectorFieldName, $vectors, $this->table, $this->vectorFieldName, $vectors,
+        ), [
+            'Surreal-NS' => $this->namespace,
+            'Surreal-DB' => $this->database,
+            'Authorization' => \sprintf('Bearer %s', $authenticationToken),
+        ]);
+
+        return array_map($this->convertToVectorDocument(...), $results[0]['result']);
+    }
+
+    public function initialize(array $options = []): void
+    {
+        $authenticationToken = $this->authenticate($options);
+
+        $this->request('POST', 'sql', \sprintf(
+            'DEFINE INDEX %s_vectors ON %s FIELDS %s MTREE DIMENSION %d DIST %s TYPE F32',
+            $this->table, $this->table, $this->vectorFieldName, $this->embeddingsDimension, $this->strategy
+        ), [
+            'Surreal-NS' => $this->namespace,
+            'Surreal-DB' => $this->database,
+            'Authorization' => \sprintf('Bearer %s', $authenticationToken),
+        ]);
+    }
+
+    /**
+     * @param array<string, mixed>|string $payload
+     * @param array<string, mixed>        $extraHeaders
+     *
+     * @return array<string|int, mixed>
+     */
+    private function request(string $method, string $endpoint, array|string $payload, array $extraHeaders = []): array
+    {
+        $url = \sprintf('%s/%s', $this->endpointUrl, $endpoint);
+
+        $finalPayload = [
+            'json' => $payload,
+        ];
+
+        if (\is_string($payload)) {
+            $finalPayload = [
+                'body' => $payload,
+            ];
+        }
+
+        $response = $this->httpClient->request($method, $url, array_merge($finalPayload, [
+            'headers' => array_merge($extraHeaders, [
+                'Accept' => 'application/json',
+                'Content-Type' => 'application/json',
+            ]),
+        ]));
```

Comment:
what about changing the `array $extraHeaders` argument to `string $authenticationToek` and moving everything to here?

```suggestion
        $response = $this->httpClient->request($method, $url, array_merge($finalPayload, [
            'headers' => [
                'Accept' => 'application/json',
                'Content-Type' => 'application/json',
                'Surreal-NS' => $this->namespace,
                'Surreal-DB' => $this->database,
                'Authorization' => \sprintf('Bearer %s', $authenticationToken),
            ],
        ]));
```

Reply:
okay, missed that, than please let's decouple those two types of requests.

meaning, instead of using the `->request(...)` in authenticate - just use the plain httpClient.

this will also remove that `is_string ? body : json` handling in the `request` method.

Reply:
The check on `string` vs `array` must stay, the SQL query is sent using `body` and not `json` 🙂